iss/src/actions/index.js

```diff
@@ -0,0 +1,52 @@
+import { getCurrentISSLocation } from '../service/issAPI'
+
+export const REQUEST_ISS_LOCATION = 'REQUEST_ISS_LOCATION';
+export const REQUEST_ISS_LOCATION_SUCCESS = 'REQUEST_ISS_LOCATION_SUCCESS';
+
+
+// Actions retornam objetos
+const requestISSLocation = () => ({
+  type: REQUEST_ISS_LOCATION
+})
+// Actions retornam objetos
+const receiveISSLocationSuccess = ({ iss_position: { latitude, longitude } }) => ({
+  type: REQUEST_ISS_LOCATION_SUCCESS,
+  latitude: parseFloat(latitude),
+  longitude: parseFloat(longitude)
+})
+
+// Action Assincrono Creator Retorna uma Funcao Com o Thunk
+export function fetchISSLocation() {
+  return (dispatch) => {
+    dispatch(requestISSLocation())
+
+    return getCurrentISSLocation()
+      .then(
+        (location) => dispatch(receiveISSLocationSuccess(location))
+      )
+  }
+}
```

Comment:
O que me impede de fazer o dispatch dessa forma
```suggestion
export function fetchISSLocation(dispatch) {
    dispatch(requestISSLocation())

    return getCurrentISSLocation()
      .then(
        (location) => dispatch(receiveISSLocationSuccess(location))
      )
}
```
e então...

Reply:
Então @EduardoSantosF, já respondendo em relação ao comentário abaixo também, nada lhe impede de usar desta forma, porém seguindo a documentação e a ideia por trás do **redux** o fluxo é sempre o **dispatch** receber a _action creator_.

Dessa forma você so modifica a sintaxe, para a função receber o metódo de dispatch mas o **redux-thunk** espera que**_async creator**_ aplique o conceito de [thunk](https://daveceddia.com/what-is-a-thunk/) que é uma função retornada por outra!

Aconselho sempre seguir padrão da documentação ah não ser que aja um motivo especifico, para alterar a propria

Reply:
ja esperava por uma resposta assim, mas para mim ainda não faz sentido algum, pq devo utilizar código a mais dar um rodeio maior sendo que o objetivo está logo ali